pkg/controllers/customeradmin/rolebindings/rolebindings.go

```diff
@@ -0,0 +1,97 @@
+package common
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	// RoleBindingNames holds the customer-admin desired rolebindings
+	RoleBindingNames = map[string]rbacv1.RoleBinding{
+		"customer-admin": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "customer-admin",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: "Group",
+					Name: "customer-admins",
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: "admin",
+			},
+		},
+		"customer-admin-project": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "customer-admin-project",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: "Group",
+					Name: "customer-admins",
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: "customer-admin-project",
+			},
+		},
+		"customer-view": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "customer-view",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: "Group",
+					Name: "customer-readers",
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: "view",
+			},
+		},
+	}
+
+	// RoleBindingNames holds the customer rolebindings
+	/*RoleBindingNames = []string{
+		"customer-admin",
+		"customer-admin-project",
+		"customer-view",
+	}*/
+
+	/*rxCustomerRoleBinding = regexp.MustCompile(`(?i)^` +
+		`customer-(admin|view|admin-project)$`)
+
+	rxRestrictedNamespace = regexp.MustCompile(`(?i)^` +
+		`(kube|` +
+		`openshift|` +
+		`default|` +
+		`kubernetes)-?`)
+	*/
+)
+
+// RoleBindingAsExpected takes the existing rolebinding and the proposed
+// and verifies that the subjects that are in the proposed exist in the
+// existing rolebinding
+func RoleBindingAsExpected(incoming, existingRoleBinding rbacv1.RoleBinding) bool {
+	if incoming.RoleRef.Name == existingRoleBinding.RoleRef.Name &&
```

Comment:
do `incoming.RoleRef == existingRoleBinding.RoleRef` so you test all fields (you were missing APIGroup).
Also, do a `reflect.DeepEqual(incoming.Subjects, existingRoleBinding.Subjects)` (similar problems).

Reply:
I did not know about `reflect.DeepEqual`.  I wondered if comparing objects would be shallow or pointer address.  Thanks!